### PR TITLE
Test for msgSV.GovCloseChannel

### PR DESCRIPTION
**Pull request by DongLieu**
## 1. Summary
Fixes #617

## 2.Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## 5. Checklist

- [ ] Does the Readme need to be updated?

### CODE REVIEW

**Review by ThanhNhann — APPROVED** [review]
lgtm! Nice work @DongLieu, I think you should check the channel is really closed in case successful with this [function](https://github.com/cosmos/ibc-go/blob/830982adfb4448573f8898bbd07a73d06bff640a/modules/apps/27-interchain-accounts/controller/keeper/keeper.go#L160)

**Comment by ThanhNhann** [comment]
And don't forget to update the branch with latest commit of branch main

**Comment by DongLieu** [comment]
The function you mentioned is in the main branch, but in the repo we use [v5.3.1](https://github.com/cosmos/ibc-go/blob/v5.3.1/modules/apps/27-interchain-accounts/controller/keeper/keeper.go#L160) it doesn't have that function

**Comment by ThanhNhann** [comment]
You can totally use the logic implemented in the function I mentioned to check that

**Comment by DongLieu** [comment]
I added a check the channel is actually closed

File: x/interchainstaking/keeper/msg_server_test.go [hunk 1]
```diff
@@ -493,3 +498,89 @@ func (suite *KeeperTestSuite) TestSignalIntent() {
 		})
 	}
 }
+
+func (suite *KeeperTestSuite) TestGovCloseChannel() {
+	testCase := []struct {
+		name     string
+		malleate func(suite *KeeperTestSuite) *icstypes.MsgGovCloseChannel
+		expecErr error
```
Comment (joe-bowman) [inline x/interchainstaking/keeper/msg_server_test.go:506]: should be 'expect', but not a blocker.

**Review by joe-bowman — APPROVED** [review]
lgtm

File: x/interchainstaking/keeper/msg_server_test.go (hunk 1 above)
Comment (joe-bowman) [inline x/interchainstaking/keeper/msg_server_test.go:506]:
```suggestion
		expectErr error
```

File: x/interchainstaking/keeper/msg_server_test.go [hunk 2]
```diff
@@ -493,3 +498,89 @@ func (suite *KeeperTestSuite) TestSignalIntent() {
...
+		{
+			name: "invalid authority",
+			malleate: func(suite *KeeperTestSuite) *icstypes.MsgGovCloseChannel {
+				return &icstypes.MsgGovCloseChannel{
+					ChannelId: "",
+					PortId:    "",
+					Authority: testAddress,
+				}
+			},
+			expecErr: govtypes.ErrInvalidSigner,
```
Comment (joe-bowman) [inline x/interchainstaking/keeper/msg_server_test.go:517]:
```suggestion
			expectErr: govtypes.ErrInvalidSigner,
```

File: x/interchainstaking/keeper/msg_server_test.go [hunk 3]
```diff
@@ -493,3 +498,89 @@ func (suite *KeeperTestSuite) TestSignalIntent() {
...
+				k := suite.GetQuicksilverApp(suite.chainA).InterchainstakingKeeper
+				channels := suite.GetQuicksilverApp(suite.chainA).IBCKeeper.ChannelKeeper.GetAllChannels(ctx)
+
+				return &icstypes.MsgGovCloseChannel{
+					ChannelId: channels[0].ChannelId,
+					PortId:    channels[0].PortId,
+					Authority: sdk.MustBech32ifyAddressBytes(sdk.GetConfig().GetBech32AccountAddrPrefix(), k.AccountKeeper.GetModuleAddress(govtypes.ModuleName)),
+				}
+			},
+			expecErr: connectiontypes.ErrInvalidConnectionState,
```
Comment (joe-bowman) [inline x/interchainstaking/keeper/msg_server_test.go:545]:
```suggestion
			expectErr: connectiontypes.ErrInvalidConnectionState,
```

File: x/interchainstaking/keeper/msg_server_test.go [hunk 4]
```diff
@@ -493,3 +498,89 @@ func (suite *KeeperTestSuite) TestSignalIntent() {
...
+				k := suite.GetQuicksilverApp(suite.chainA).InterchainstakingKeeper
+				channels := suite.GetQuicksilverApp(suite.chainA).IBCKeeper.ChannelKeeper.GetAllChannels(ctx)
+
+				return &icstypes.MsgGovCloseChannel{
+					ChannelId: channels[0].ChannelId,
+					PortId:    channels[0].PortId,
+					Authority: sdk.MustBech32ifyAddressBytes(sdk.GetConfig().GetBech32AccountAddrPrefix(), k.AccountKeeper.GetModuleAddress(govtypes.ModuleName)),
+				}
+			},
+			expecErr: nil,
```
Comment (joe-bowman) [inline x/interchainstaking/keeper/msg_server_test.go:561]:
```suggestion
			expectErr: nil,
```

File: x/interchainstaking/keeper/msg_server_test.go [hunk 5]
```diff
@@ -493,3 +498,89 @@ func (suite *KeeperTestSuite) TestSignalIntent() {
...
+		suite.Run(tc.name, func() {
+			suite.SetupTest()
+			suite.setupTestZones()
+
+			msg := tc.malleate(suite)
+			msgSrv := icskeeper.NewMsgServerImpl(*suite.GetQuicksilverApp(suite.chainA).InterchainstakingKeeper)
+			ctx := suite.chainA.GetContext()
+
+			_, err := msgSrv.GovCloseChannel(ctx, msg)
+			if tc.expecErr != nil {
```
Comment (joe-bowman) [inline x/interchainstaking/keeper/msg_server_test.go:574]:
```suggestion
			if tc.expectErr != nil {
```

File: x/interchainstaking/keeper/msg_server_test.go [hunk 6]
```diff
@@ -493,3 +498,89 @@ func (suite *KeeperTestSuite) TestSignalIntent() {
...
+				suite.ErrorIs(tc.expecErr, err)
```
Comment (joe-bowman) [inline x/interchainstaking/keeper/msg_server_test.go:575]:
```suggestion
				suite.ErrorIs(tc.expectErr, err)
```

**Review by ThanhNhann — APPROVED** [review]

**Review by joe-bowman — APPROVED** [review]
lgtm

**Pull request merged**